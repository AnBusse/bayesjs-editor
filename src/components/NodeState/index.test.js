import React from 'react';
import { shallow } from 'enzyme';
import NodeState from './component';

const defaultProps = {
  state: 'state',
  onStateDoubleClick: jest.fn(),
  results: {},
  y: 5,
  height: 20,
  textX: 5,
  barX: 80,
  textWidth: 70,
  barWidth: 70,
};

const shallowComponent = (props = {}) => {
  const compProps = { ...defaultProps, ...props };

  return shallow(<NodeState {...compProps} />);
};

describe('NodeState Component', () => {
  it('matches snapshot', () => {
    const component = shallowComponent();

    expect(component).toMatchSnapshot();
  });

  describe('Event Listeners', () => {
    describe('onStateDoubleClick', () => {
      const onStateDoubleClickInnerEvent = jest.fn();
      const onStateDoubleClick = jest.fn(() => onStateDoubleClickInnerEvent);
      const component = shallowComponent({ onStateDoubleClick });

      beforeAll(() => {
        component.find('NodeStateBar').simulate('doubleClick');
      });

      it('triggers "onStateDoubleClick" render component', () => {
        expect(onStateDoubleClick).toBeCalledWith('state');
      });

      it('triggers "onStateDoubleClickInnerEvent" on clicked NodeStateBar', () => {
        expect(onStateDoubleClickInnerEvent).toBeCalled();
      });
    });
  });
});
