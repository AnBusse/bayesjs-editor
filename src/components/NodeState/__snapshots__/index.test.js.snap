// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeState Component matches snapshot 1`] = `
<g
  data-testid="node-state__state"
>
  <SvgText
    height={20}
    text="state"
    width={70}
    x={5}
    y={5}
  />
  <NodeStateBar
    belief={null}
    height={20}
    results={Object {}}
    state="state"
    width={70}
    x={80}
    y={5}
  />
</g>
`;
